### PR TITLE
Issue 1943 allow jscode in options

### PR DESCRIPTION
**Pull request by hansthen**
Boring but big. The leaflet class hierarchy allows all classes to have optional arguments. These optional arguments can contain JavaScript code or references to other Leaflet elements.

To support this in Folium I replaced (almost) all instances of `parse_options` in the constructors with `this.options | tojavascript` in the templates. I left the `parse_options` in case someone outside of Folium uses it.

### CODE REVIEW

File: folium/plugins/realtime.py [hunk 1]
```diff
@@ -124,12 +113,6 @@ def __init__(
         if update_feature is not None:
             kwargs["update_feature"] = JsCode(update_feature)
         kwargs["remove_missing"] = remove_missing
+        kwargs["container"] = container
```
Comment (Conengmo) [inline folium/plugins/realtime.py:116]:
```suggestion
        kwargs["container"] = JsCode(container.get_name())
```

You  made this class right, so you probably know better. But I think it needs the Javascript name as JS code.

File: folium/utilities.py [hunk 2]
```diff
@@ -365,7 +365,8 @@ def temp_html_filepath(data: str) -> Iterator[str]:
         yield filepath
     finally:
         if os.path.isfile(filepath):
-            os.remove(filepath)
+            pass
+            # os.remove(filepath)
```
Comment (Conengmo) [inline folium/utilities.py:369]: Guess this is an artefact from testing? Should be restored.

```suggestion
            os.remove(filepath)
```

File: tests/plugins/test_grouped_layer_control.py [hunk 3]
```diff
@@ -34,10 +35,11 @@ def test_grouped_layer_control():
...
-            {{"exclusiveGroups": ["groups1"]}},
+            {{"exclusiveGroups": ["groups1",],}},
          ).addTo({m.get_name()});
          {fg2.get_name()}.remove();
     """
-        )
-        in out
     )
+    print(out)
+    print(expected)
```
Comment (Conengmo) [inline tests/plugins/test_grouped_layer_control.py:44]:
```suggestion
```

File: tests/plugins/test_marker_cluster.py [hunk 4]
```diff
@@ -61,6 +61,8 @@ def test_marker_cluster():
         in out
     )  # noqa
 
+    print(out)
+    print(expected)
```
Comment (Conengmo) [inline tests/plugins/test_marker_cluster.py:65]:
```suggestion
```

File: tests/test_folium.py [hunk 5]
```diff
@@ -464,28 +464,31 @@ def test_global_switches(self):
         m = folium.Map(prefer_canvas=True)
         out = m._parent.render()
         out_str = "".join(out.split())
-        assert "preferCanvas:true" in out_str
+        for line in out.split("\n"):
+            print(line)
+        print(out_str)
```
Comment (Conengmo) [inline tests/test_folium.py:469]:
```suggestion
```

File: tests/test_raster_layers.py [hunk 6]
```diff
@@ -65,6 +65,7 @@ def test_wms():
     )
     w.add_to(m)
     html = m.get_root().render()
+    print(html)
```
Comment (Conengmo) [inline tests/test_raster_layers.py:68]:
```suggestion
```

**Review by Conengmo — CHANGES_REQUESTED** [review]
Boring but big 😄 Kudos for taking this on. It looks good! I have two small comments, plus some suggestions to remove added print statements from tests. Other than that looks good!

File: folium/plugins/realtime.py (hunk 1 above)
Comment (hansthen) [reply folium/plugins/realtime.py:116]: We test specifically for Branca `Element`s in `tojavascript` so it's not necessary. 
```     
elif isinstance(obj, Element):
        return obj.get_name()
```

Comment (Conengmo) [reply folium/plugins/realtime.py:116]: Nice!

**Review by Conengmo — APPROVED** [review]
Let's go! 🚀

**Comment by hansthen** [comment]
@Conengmo Perhaps you can also have a look at the `raster_layers.WmsTileLayer` class. I still use `{{ this.options|tojson }}` there because there is an option that should remain lowercase. The inconsistency bugs me (a bit). I could not think of a way to make it consistent. But perhaps you have suggestions on how we could use `tojavascript` here?

**Comment by Conengmo** [comment]
Yes, I noticed that one as well. Can't think of a good, simple solution right now, so maybe what you did is best for now.

**Pull request merged**